iscsi/iscsi.go

```diff
@@ -211,21 +203,21 @@ func pathExists(devicePath *string, deviceTransport string) error {
 }
 
 // getMultipathDevice returns a multipath device for the configured targets if it exists
-func getMultipathDevice(devices []Device) (*Device, error) {
+func getMultipathDevice(logger *logr.Logger, devices []Device) (*Device, error) {
 	var multipathDevice *Device
 
+	// This routine relies on output from lsblk, and older versions did not produce the desired outcome
+	// of a child row, followed by a parent row. This routine now just takes the first parent found.
 	for _, device := range devices {
+		logger.V(1).Info("find multipath device", "device", device)
 		if len(device.Children) != 1 {
-			multipathdNotRunning := ""
-			if len(device.Children) == 0 {
-				multipathdNotRunning = " (is multipathd running?)"
-			}
-			return nil, fmt.Errorf("device is not mapped to exactly one multipath device%s: %v", multipathdNotRunning, device.Children)
+			logger.V(1).Info("WARNING: children != 1", "name", device.Name)
 		}
-		if multipathDevice != nil && device.Children[0].Name != multipathDevice.Name {
-			return nil, fmt.Errorf("devices don't share a common multipath device: %v", devices)
+		if len(device.Children) == 1 {
```

Comment:
Arent we missing the check of device->chilidren matching against multipathDevice here ?

Reply:
As a hopefully more acceptable solution, I added a new optional parameter that would allow the code to execute as it did before this change, or work with the lsblk output that lists only one parent instead of a parent for each child. I verified that the make test works and also tested with a broken version of lsblk, creating 200 PVCs.